File: library/src/main/java/me/proxer/library/entity/info/EntrySeasonInfo.java
package me.proxer.library.entity.info;

import com.squareup.moshi.Json;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Value;
import me.proxer.library.entity.ProxerIdItem;
import me.proxer.library.enums.Season;

import javax.annotation.Nullable;

/**
 * Entity holding the season of an entry.
 *
 * @author Ruben Gees
 */
@Value
@EqualsAndHashCode(onParam = @__({@Nullable}))
public class EntrySeasonInfo implements ProxerIdItem {

    /**
     * Returns the type.
     */
    @Getter(onMethod = @__({@Override}))
    @Json(name = "id")
    private String id;

    /**
     * Returns the year.
     */
    @Json(name = "year")
    private int year;

    /**
     * Returns the actual season.
     */
    @Json(name = "season")
    private Season season;
}

File: library/src/main/kotlin/me/proxer/library/entity/info/EntrySeasonInfo.kt
package me.proxer.library.entity.info

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass
import me.proxer.library.entity.ProxerIdItem
import me.proxer.library.enums.Season

/**
 * Entity holding the season of an entry.
 *
 * @property year The year.
 * @property season The actual season.
 *
 * @author Ruben Gees
 */
@JsonClass(generateAdapter = true)
data class EntrySeasonInfo(
    @Json(name = "id") override val id: String,
    @Json(name = "year") val year: Int,
    @Json(name = "season") val season: Season
) : ProxerIdItem
